pkg/engine/doc.go

```diff
@@ -0,0 +1,12 @@
+// Package engine holds code that drive the engine of Kusion.
+//
+// Kusion Engine is a middle layer between Spec and the actual infrastructure.
+// The major function of this engine is to parse the KCL compiled result and to turn all actual infra resources into the desired state
+// described in KCL. It consists of 3 parts:
+// 1. DAG engine: parse resources in the Spec and figure out which resource should be created,
+// updated or deleted. The analysis result will be converted into a DAG.
+// 2. Runtime: an interface that represents an actual infrastructure.
+// All operations that trying to manipulate a resource should delegate to one Runtime
+// 3. State: state is a record of an operation's result.
+// It is often used as a datasource for 3-way merge/diff in operations like Apply or Preview
```

Comment:
+ Kusion Engine is a middle layer between API spec and the actual infrastructure. And Who will use the Engine?
+ Remove 'KCL' in the doc.  API spec does not need to perceive the upper layer input.
+ API Spec should be versioned, and should be backwards compatible?
+ The conversion mechanism, codec, convention.. from api spec to internal model?
+ I prefer to split 'DAG engine' to a diff engine, a DAG builder and an execution engine?
+ Runtime: runtime is usually not an interface 
+ Important interface, structs and usage examples?

Reply:
Judging from the code I have learned so far and his positioning, maybe this code structure would be more appropriate:

/types -- engine‘s internal types
/models -- engine's internal model
/exec -- nodes exec flow
/dag -- dag builder, walker, etc
/diff -- diff tools
/runtime -- runtime apis and impls
/state -- state apis and impls
api.go -- engine's core public interfaces and functionalities.
types.go -- entine's public types.

Through such code organization, the core functions of the engine can be better showed, and the codes of different functional modules can also be reasonably organized.

You can modify it as you like. Hope the advice helps.